tests/unit/browser/test_caret.py

```diff
@@ -0,0 +1,364 @@
+        caret.move_to_prev_char()
+        selection.toggle()
+        caret.move_to_end_of_word()
+        selection.check("e")
+
+
+def test_drop_selection(caret, selection):
+    selection.toggle()
+    caret.move_to_end_of_word()
+    caret.drop_selection()
+    selection.check("")
+
+
+class TestSearch:
+
+    # https://bugreports.qt.io/browse/QTBUG-60673
+
+    @pytest.mark.qtbug60673
+    @pytest.mark.no_xvfb
+    def test_yanking_a_searched_line(self, caret, selection, mode_manager, callback_checker, web_tab, qtbot):
+        web_tab.show()
+        mode_manager.leave(usertypes.KeyMode.caret)
+
+        web_tab.search.search('fiv', result_cb=callback_checker.callback)
+        callback_checker.check(True)
+
+        mode_manager.enter(usertypes.KeyMode.caret)
+        caret.move_to_end_of_line()
+        selection.check('five six')
+
+    @pytest.mark.qtbug60673
+    @pytest.mark.no_xvfb
+    def test_yanking_a_searched_line_with_multiple_matches(self, caret, selection, mode_manager, callback_checker, web_tab, qtbot):
+        web_tab.show()
+        mode_manager.leave(usertypes.KeyMode.caret)
+
+        web_tab.search.search('w', result_cb=callback_checker.callback)
+        callback_checker.check(True)
+
+        web_tab.search.next_result(result_cb=callback_checker.callback)
+        callback_checker.check(True)
+
+        mode_manager.enter(usertypes.KeyMode.caret)
+
+        caret.move_to_end_of_line()
+        selection.check('wei drei')
+
+
+class TestFollowSelected:
+
+    LOAD_STARTED_DELAY = 50
+
+    @pytest.fixture(params=[True, False], autouse=True)
+    def toggle_js(self, request, config_stub):
+        config_stub.val.content.javascript.enabled = request.param
+
+    def test_follow_selected_without_a_selection(self, qtbot, caret, selection, web_tab,
+                                                 mode_manager):
+        caret.move_to_next_word()  # Move cursor away from the link
+        mode_manager.leave(usertypes.KeyMode.caret)
+        with qtbot.wait_signal(caret.follow_selected_done):
+            with qtbot.assert_not_emitted(web_tab.load_started):
+                caret.follow_selected()
+                qtbot.wait(self.LOAD_STARTED_DELAY)
+
+    def test_follow_selected_with_text(self, qtbot, caret, selection, web_tab):
+        caret.move_to_next_word()
+        selection.toggle()
+        caret.move_to_end_of_word()
+        with qtbot.wait_signal(caret.follow_selected_done):
+            with qtbot.assert_not_emitted(web_tab.load_started):
+                caret.follow_selected()
+                qtbot.wait(self.LOAD_STARTED_DELAY)
+
+    def test_follow_selected_with_link(self, caret, selection, config_stub,
+                                       qtbot, web_tab):
+        selection.toggle()
+        caret.move_to_end_of_word()
+        with qtbot.wait_signal(web_tab.load_finished):
+            caret.follow_selected()
+        assert web_tab.url().path() == '/data/hello.txt'
```

Comment:
Maybe this is out of scope for this PR, but it would be nice to pull the search `follow-selected` tests into unit tests as well (and maybe put them here). If you're busy, I can help with that.

Reply:
I'd like to convert (almost) all tests to unit tests in the long run, just thought I'd start with some low-hanging fruit and see how it goes. Turns out, it's more annoying than expected...